### ISSUE TITLE
[Feature] Use Dark Theme by Default

### ISSUE THREAD

**Issue by Char2sGu** [ng-doc/ng-doc#149]
### Description

When using SSR, the rendered page is initially displayed in the default light theme. If the user's OS uses a dark palette and the theme switching is based on the OS settings, this would cause a dazzling white flash on page load.

Imagine this: A developer is sitting in a dark room focusing on learning a library. When he clicks into the documentation website powered by NgDoc, the screen suddenly turns into a flashbang.

### Proposed solution

I believe rendering a dark theme by default would harm less, since starting from a dark theme and then turn into a light theme could be less "surprising" than the opposite way.

### Alternatives considered

Or let's change the definition of a "theme". A theme must include two color schemes: light and dark. By default, the color scheme follows the `prefers-color-scheme` query, and after the hydration is finished, the application can apply additional classes on the `<html>` or `<body>` element to force a color scheme.

**Comment by skoropadas** [comment]
This is a known bug that appeared with the transition to SSR. It will be addressed within another ticket (#126 which includes a discussion of this issue), so I'll close this one as a duplicate.

As for the default theme, I think I'll leave everything as is to avoid sudden breaking changes. In any case, with the bug fix, there shouldn't be any more issues with the dark theme.

**Comment by Char2sGu** [comment]
Glad to know the default theme fix is already in progress.

However, what do you think of the first screen before hydration?  
It is theoretically not possible to know the user's `prefer-color-scheme` at the server side. If we render one theme by default and then change to a dark theme based on the user setting after hydration, there will definitely be a flash.  
My proposal is to combine the light scheme and dark scheme into one theme. The scheme to display is determined by `prefer-color-scheme` by default, but can be explicitly specified through html classes after hydration.
In other words, we split the concept of color scheme and theme. A theme consists of two color schemes: light and dark.

**Comment by skoropadas** [comment]
Switching to the dark theme by default still doesn't solve the problem. Before transitioning to SSR, the necessary theme was loaded via HTTP during the application initialization stage, and there was no flash. Now, it occurs more because the browser first renders what the server sends and then executes scripts. Therefore, if, for example, you move this logic into a separate script and include it in `index.html`, there won't be any flash.

So I think the only solution could be integrating a script inside `index.html` that sets the required theme, similar to [what angular.dev does](https://github.com/angular/angular/blob/a84cb62522bff516e3b05eaa7b7ff9a0ca46c561/adev/src/index.html#L9). But anyways, I'll consider this and look for solutions that would fix this flash.

**Comment by Char2sGu** [comment]
Would you like me to create a new ticket for that?

**Comment by skoropadas** [comment]
> Would you like me to create a new ticket for that?

There's no need for that, I'll do everything when I start working on the #126
